### PR TITLE
make multi target and intermediate SConf files marked is_conftest

### PR DESCRIPTION
**Pull request by dmoody256**
Discovered some issues with is_conftest when testing mongo build with experimental ninja tool.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: SCons/Builder.py [hunk 1]
```diff
@@ -613,6 +613,10 @@ def _execute(self, env, target, source, overwarn={}, executor_kw={}):
             t.set_executor(executor)
             t.set_explicit(self.is_explicit)
 
+        if env.get("SCONF_NODE"):
+            for node in tlist + slist:
+                node.attributes.conftest_node = 1
```
Comment (dmoody256) [inline SCons/Builder.py:618]: Thinking about this more, is there a case where a user would have non conftest file as an input to a conftest? maybe the slist should not be included in this loop?

Comment (dmoody256) [reply SCons/Builder.py:618]: I added slist to make sure intermediates were marked but, the execute will be called recursively to build intermediates so those intermediates should be targets in the tlist at some point.

File: SCons/SConf.py [hunk 2]
```diff
@@ -733,9 +736,6 @@ def _createDir( self, node ):
             if not os.path.isdir( dirName ):
                 os.makedirs( dirName )
 
-    def _set_conftest_node(self, node):
```
Comment (dmoody256) [inline SCons/SConf.py:736]: no reason for this to be part of the class and muck around with self arg, moved it to an isolated function

File: test/Configure/is_conftest/fixture/SConstruct [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    conf.env.Append(
+        CCFLAGS="/DEBUG /Z7 /INCREMENTAL:NO",
+        LINKFLAGS="/DEBUG /INCREMENTAL:NO",
+        PDB='${TARGET.base}.pdb')
+if not conf.TryRun("int main( int argc, char* argv[] ){return 0;}", '.c'):
+    print("FAIL")
+
+env = conf.Finish()
+
+for node in env.Glob('.sconf_temp/*'):
```
Comment (dmoody256) [inline test/Configure/is_conftest/fixture/SConstruct]: probably should set the conf_dir explicitly to make this test a bit more future proof?

File: SCons/SConf.py (hunk 2 above)
Comment (mwichmann) [reply SCons/SConf.py:736]: Or @staticmethod

File: test/Configure/is_conftest/fixture/SConstruct (hunk 3 above)
Comment (bdbaddog) [reply test/Configure/is_conftest/fixture/SConstruct]: won't this fail for the .ilk files we discussed previously?

File: test/Configure/is_conftest/is_conftest.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+#
+# __COPYRIGHT__
```
Comment (bdbaddog) [inline test/Configure/is_conftest/is_conftest.py]: Please use the new copyright blurb..   (you can see it in `template/test.py`)

File: test/Configure/is_conftest/fixture/SConstruct (hunk 3 above)
Comment (bdbaddog) [reply test/Configure/is_conftest/fixture/SConstruct]: You can get the config dir from conf via `conf.confdir` for use in your Glob..

File: SCons/Builder.py (hunk 1 above)
Comment (bdbaddog) [reply SCons/Builder.py:618]: Yes. I think tlist should be sufficient.  slist could include header files external to the build which could change and should cause a rebuild?

File: test/Configure/is_conftest/fixture/SConstruct (hunk 3 above)
Comment (dmoody256) [reply test/Configure/is_conftest/fixture/SConstruct]: > won't this fail for the .ilk files we discussed previously?

incremental linking is set to NO, so msvc wont generate ilk files.

File: SCons/Builder.py (hunk 1 above)
Comment (dmoody256) [reply SCons/Builder.py:618]: slist removed

File: test/Configure/is_conftest/fixture/SConstruct (hunk 3 above)
Comment (dmoody256) [reply test/Configure/is_conftest/fixture/SConstruct]: updated to use conf.confdir

File: test/Configure/is_conftest/is_conftest.py (hunk 4 above)
Comment (dmoody256) [reply test/Configure/is_conftest/is_conftest.py]: updated to the template copyright

File: test/Configure/is_conftest/fixture/SConstruct (hunk 3 above)
Comment (bdbaddog) [reply test/Configure/is_conftest/fixture/SConstruct]: > updated to use conf.confdir

Did you push that?

Comment (mwichmann) [reply test/Configure/is_conftest/fixture/SConstruct]: It's also an environment var, no?  https://scons.org/doc/production/HTML/scons-man.html#cv-CONFIGUREDIR

Comment (dmoody256) [reply test/Configure/is_conftest/fixture/SConstruct]: > > updated to use conf.confdir
> 
> Did you push that?

yeah f21e361

**Pull request merged**